src/main/kotlin/exercise4/task1/BankAccount.kt

```diff
@@ -33,20 +33,51 @@ package exercise4.task1
  */
 
 
+open class BankAccount(
+    private val accountNumber: String,
+    private val accountHolderName: String,
+    private var balance: Double = 0.0
+) {
+    open fun deposit(amount: Double) {
+        balance += amount;
+    }
+
+    open fun withdraw(amount: Double): Boolean {
+        if (balance - amount >= 0) {
+            balance -= amount
+            return true
+        }
+        return false;
+    }
+
+    fun getBalance(): Double {
```

Comment:
Another option would be to annotate the `balance` property with a `protected` modifier. That way, property would be visible in a derived class but not outside of the original and derived class.